### PR TITLE
Remove Metrics Namespace API : Part 1

### PR DESCRIPTION
**Pull request by jakelandis**
This is part of a much much larger commit set kept small for easier review.

The tests are not expected to pass, and will fixup the commit log once all parts are reviewed.

Changes in this commit are:
* Removal of Metrics namespace and related Ruby classes.
* Changes neccessary for the agent.rb class
* Removal of un-dcoumented optional 'turn off metrics' flag

Part of #7788

### CODE REVIEW

File: logstash-core/lib/logstash/agent.rb [hunk 1]
```diff
@@ -464,51 +436,27 @@ def update_metrics(converge_result)
...
-      n.gauge(:last_error, { :message => action_result.message, :backtrace => action_result.backtrace})
-      n.gauge(:last_failure_timestamp, LogStash::Timestamp.now)
-    end
-  end
 
-  def initialize_agent_metrics
-    @instance_reload_metric.increment(:successes, 0)
-    @instance_reload_metric.increment(:failures, 0)
-  end
+    @witness_reloads.failure
```
Comment (andrewvc) [inline logstash-core/lib/logstash/agent.rb]: ❤️  the ratio of removals to deletions here

File: logstash-core/lib/logstash/agent.rb [hunk 2]
```diff
@@ -464,51 +436,27 @@ def update_metrics(converge_result)
...
-      n.increment(:failures, 0)
-      n.gauge(:last_error, nil)
-      n.gauge(:last_success_timestamp, nil)
-      n.gauge(:last_failure_timestamp, nil)
-    end
+    witness_pipeline_reloads = @witness_pipelines.pipeline(action.pipeline_id).reloads
+    witness_pipeline_reloads.failure
+    witness_pipeline_reloads.error.message(action_result.message)
+    witness_pipeline_reloads.error.backtrace(action_result.backtrace.to_s)
+    witness_pipeline_reloads.last_failure_timestamp(LogStash::Timestamp.now)
```
Comment (andrewvc) [inline logstash-core/lib/logstash/agent.rb]: Can we consolidate these operations into a single method on the witness? My thinking is that it'd be a cleaner separation of concerns to have the externally used witness methods correspond to lifecycle events rather than act as setters.

Maybe something like:

```ruby
witness.pipeline(action.pipeline_id).failure(message, action_result.backtrace.to_s)
```

File: logstash-core/lib/logstash/agent.rb [hunk 3]
```diff
@@ -464,51 +436,27 @@ def update_metrics(converge_result)
...
 
   def update_successful_reload_metrics(action, action_result)
-    @instance_reload_metric.increment(:successes)
+    @witness_reloads.success
 
-    @pipeline_reload_metric.namespace([action.pipeline_id, :reloads]).tap do |n|
-      n.increment(:successes)
-      n.gauge(:last_success_timestamp, action_result.executed_at)
-    end
+    witness_pipeline_reloads = @witness_pipelines.pipeline(action.pipeline_id).reloads
```
Comment (andrewvc) [inline logstash-core/lib/logstash/agent.rb]: Same note RE: consolidation of operations

File: logstash-core/spec/logstash/agent_spec.rb [hunk 4]
```diff
@@ -329,7 +329,6 @@
 
     let(:agent_args) do
       {
-        "metric.collect" => true,
```
Comment (andrewvc) [inline logstash-core/spec/logstash/agent_spec.rb]: Why was this removed? Were metrics off by default during tests before?

Comment (jakelandis) [reply logstash-core/spec/logstash/agent_spec.rb]: moving forward, there will not be any 'turn on/off the metrics' flag

File: logstash-core/lib/logstash/agent.rb (hunk 2 above)
Comment (jakelandis) [reply logstash-core/lib/logstash/agent.rb]: +1 , i will gather these all suggestions like this up over at https://github.com/elastic/logstash/issues/7973 and make a final sweep of the code with those changes.

**Comment by jakelandis** [comment]
I moved https://github.com/elastic/logstash/issues/7973 upto the 'TODO' list before #7788 is considered done. (which accounts for the consolidation referenced above). 

Merging to feature branch.
